internal/migrations/changeset_listener.go

```diff
@@ -0,0 +1,272 @@
+				continue
+			}
+			btsReceived := int64(0)
+
+			service.Logger.Debug("received changeset metadata", log.Int("height", currentHeight), log.Int("numChunks", numChunks), log.Int("size", size))
+			wg := sync.WaitGroup{}
+			errChan := make(chan error, numChunks)
+
+			for i := int64(0); i < numChunks; i++ {
+				wg.Add(1)
+				go func(chunkIdx int64) {
+					for {
+						// Load the changeset from the admin server
+						cs, err := adminClient.LoadChangeset(ctx, currentHeight, chunkIdx)
+						if err != nil {
+							// If no changeset is found, wait and try again
+							time.Sleep(retryDelay)
+							continue
+						}
+
+						// Create the changeset migration event
+						evt := &ChangesetMigration{
+							Height:      big.NewInt(currentHeight),
+							TotalChunks: big.NewInt(numChunks),
+							ChunkIdx:    big.NewInt(chunkIdx),
+							Changeset:   cs,
+						}
+
+						csEvt, err := evt.MarshalBinary()
+						if err != nil {
+							service.Logger.Error("failed to marshal changeset migration event", log.String("error", err.Error()))
+							errChan <- err
+							return
+						}
+
+						service.Logger.Info("broadcasting changeset migration event", log.Int("height", currentHeight), log.Int("chunk", chunkIdx), log.Int("size", len(cs)))
+
+						// Broadcast the changeset migration event to the event store for voting
+						err = eventStore.Broadcast(ctx, ChangesetMigrationEventType, csEvt)
+						if err != nil {
+							errChan <- fmt.Errorf("failed to broadcast changeset migration event: %w", err)
+							return
+						}
+
+						btsReceived += int64(len(cs))
+						break
+					}
+					wg.Done()
+				}(i)
+			}
+
+			// Wait for all chunks to be successfully received
+			wg.Wait()
+			close(errChan)
+
+			// check errChan for all the errors
+			var errs error
+			for err := range errChan {
+				errs = errors.Join(errs, err)
+			}
+
+			if errs != nil {
+				return errs
+			}
+
+			if btsReceived != size {
+				return fmt.Errorf("incorrect changeset size received: %d, expected: %d", btsReceived, size)
+			}
+
+			if err = setLastStoredHeight(ctx, eventStore, currentHeight); err != nil {
+				return err
+			}
+			currentHeight++
+
+			// Delay before polling the admin server again for the next changeset
+			time.Sleep(retryDelay)
+		}
+	}
+}
+
+func (c *MigrationConfig) ExtractConfig(cfg map[string]string) error {
+	var err error
+	var ok bool
+
+	startHeight, ok := cfg["start_height"]
+	if !ok {
+		return errors.New("migration start_height not provided")
+	}
+	c.StartHeight, err = strconv.ParseInt(startHeight, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	endHeight, ok := cfg["end_height"]
+	if !ok {
+		return errors.New("migration end_height not provided")
+	}
+	c.EndHeight, err = strconv.ParseInt(endHeight, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	c.AdminListenAddress, ok = cfg["admin_listen_address"]
+	if !ok {
+		return errors.New("migration admin_listen_address not provided")
+	}
+
+	c.AdminPass = cfg["admin_pass"]
```

Comment:
One of the reasons I want the migration JSON RPC getter methods bound to txsvc is because it allows users to migrate without exposing their admin endpoint (which by default only runs on UNIX sockets). This makes it easier to leverage this feature. Assuming we go ahead with this change, we should update that in the oracle configuration.